### ISSUE TITLE
References to nested types in other packages are misnamed (missing .)

### ISSUE THREAD

**Issue by asfalcone** [Kotlin/dokka#2245]
**Describe the bug**
References to nested types in other packages are misnamed (missing .)

**Expected behaviour**
`PagingDataAdapter` references `androidx.recyclerview.widget.DiffUtil.ItemCallback` as type for a constructor param. The name of the type should be `DiffUtil.ItemCallback` but instead it is parsed into the model incorrectly as `DiffUtilItemCallback` which causes links to not resolve 

**Screenshots**
If applicable, add screenshots to help explain your problem
![image](https://user-images.githubusercontent.com/1848683/143079107-690b288b-7c35-41cf-bfd3-52f309289597.png)

```
abstract class PagingDataAdapter<T : Any, VH : RecyclerView.ViewHolder> @JvmOverloads constructor(
    diffCallback: DiffUtil.ItemCallback<T>,
    mainDispatcher: CoroutineDispatcher = Dispatchers.Main,
    workerDispatcher: CoroutineDispatcher = Dispatchers.Default
) : RecyclerView.Adapter<VH>() {
```

(source code: https://cs.android.com/androidx/platform/tools/dokka-devsite-plugin/+/master:testData/paging/source/androidx/paging/PagingDataAdapter.kt;l=62)

**Installation**
- Dokka version: 1.6.0-dev-128
(This is a regression since 1.6.0-dev-122).

**Comment by vmishenev** [comment]
Now on your master branch I can not find a class `DiffUtil.ItemCallback<>`
By t`estData/paging/docs/reference/kotlin/androidx/paging/PagingDataAdapter.html`  it does not resolve and renders as `ERROR CLASS`.
On practice we do not use `DRI.callable` for unresolved classes. This bug appears only for such classes in `DRI.callable`. 

But anyway I will fix it, unfortunately, it is on Kotlin compiler side. 
![image](https://user-images.githubusercontent.com/9636937/146743336-aa9553f8-d196-44e3-a051-d350a36fbb06.png)

**Comment by IgnatBeresnev** [comment]
Running tests for devsite plugin with upcoming 1.6.10 now, and it seems like either it has gotten better (after upgrading Kotlin) or I'm just lucky with my local runs. Some (if not all) tests from `com.google.devsite.integration.BasicTest` now fail because the dot **is** present in places where it wasn't at the time of generating test data.

Example
```
 - AlignmentHorizontal
 + Alignment.Horizontal

-  <p>Used to specify the vertical arrangement of the layout's children in layouts like <code><a href="/reference/androidx/compose/foundation/layout/package-summary.html#Column(Modifier,androidx.compose.foundation.layout.Arrangement.Vertical,AlignmentHorizontal,kotlin.Function1)">Column</a></code>.</p>
+  <p>Used to specify the vertical arrangement of the layout's children in layouts like <code><a href="/reference/androidx/compose/foundation/layout/package-summary.html#Column(Modifier,androidx.compose.foundation.layout.Arrangement.Vertical,Alignment.Horizontal,kotlin.Function1)">Column</a></code>.</p>
```

So it may have been fixed in the kotlin compiler (I'll try to verify it), just wanted to give you a heads up.

**Comment by asfalcone** [comment]
Appears fixed in 1.6.10-dev-139

**Comment by IgnatBeresnev** [comment]
Nice to hear! 

I'll close the issue for now, we'll reopen and investigate it further if it starts happening again. Hopefully, this has been fixed for good
